File: scripting-perf-compare/test_regex.rb
#! /usr/bin/env ruby

infilename = ARGV[0]
outfilename = ARGV[1]
r = /^(\d+)\tMember\((\d+)\)\*Name\((\d+)\)/

File.open(infilename, "r") { |infile|
  File.open(outfilename, "w") { |outfile|
    while (line = infile.gets)
      groups = r.match(line)
      fail "Failed to match line" if groups == nil
      outfile.puts "==> " + groups[1] + " " + groups[2] + " " + groups[3]
    end
  }
}

File: scripting-perf-compare/test_regex.py
#! /usr/bin/env python2

import re
import sys

infilename = sys.argv[1]
outfilename = sys.argv[2]

infile = open(infilename, 'r')
outfile = open(outfilename, 'w')
p = re.compile('^(\d+)\tMember\((\d+)\)\*Name\((\d+)\)')

for line in infile:
    line = line.rstrip("\r\n")
    m = p.match(line)
    outfile.write("==> " + m.group(1) + " " + m.group(2) + " " + m.group(3) + "\n")

infile.close()
outfile.close()
